WordPress/Classes/ViewRelated/Blog/Site Picker/SitePickerViewController.swift

```diff
@@ -93,19 +94,27 @@ extension SitePickerViewController: BlogDetailHeaderViewDelegate {
     }
 
     func siteSwitcherTapped() {
-        let blogListController = BlogListViewController(configuration: .defaultConfig, meScenePresenter: meScenePresenter)
-
-        blogListController.blogSelected = { [weak self] controller, selectedBlog in
-            guard let self else { return }
-            self.switchToBlog(selectedBlog)
-            controller.dismiss(animated: true) {
-                self.onBlogListDismiss?()
+        var dismissAction: (() -> Void)? = nil
+        let hostingController = UIHostingController(
+            rootView: SiteSwitcherView(
+                selectionCallback: { [weak self] siteID in
+                    guard let selectedBlog = BlogListViewModel().allBlogs.first(where: { $0.dotComID == siteID }) else {
+                        return
+                    }
+                    self?.switchToBlog(selectedBlog)
+                    // Dismiss hosting controller with completion block
+                    dismissAction?()
+                }, addSiteCallback: { [weak self] in
+                    self?.addSiteTapped()
```

Comment:
Why do we need the add site to be passed a block? Why can't we implement it inside `SiteSwitcherView`? The changes made in `SitePickerViewController+SiteActions` is not great for readability.

Reply:
The problem is, we'll need to invoke `SitePickerViewController` somehow as it updates the screen on which site is selected. `SiteSwitcherView` is agnostic to what happens in MySite, i.e. `SitePickerViewController`. 

`launchSiteCreation` function needed to be updated because of how the view hierarchy was set on these files. Noting the deleted line `parent?.present(actionSheet, animated: true)`. 

I prefer not to play with these files further in the context of this PR. It already handles too many things. If you think this needs to be changed, can I ask you what kind of changes do you have in mind because I don't want to do more ping pong on this PR.

Reply:
Thanks for the context! It has helped me understand the direction you took.

> I prefer not to play with these files further in the context of this PR.

My initial intuition was to stay away from these files as well.

> can I ask you what kind of changes do you have in mind because I don't want to do more ping pong on this PR.

Sure! What I'm suggesting is to create a UIHostingController subclass called `SiteSwitcherViewController` for example. In this subclass, we can implement the addSite functionality the same way it is implemented in the legacy `BlogListViewController`. The downside of my suggestion is that we will be duplicating code between the new and legacy site switchers. The upside is we won't be tampering with existing code in the site picker. (P.S.: Site picker is a terrible name for this component as it can be easily confused with the site switcher.)